src/main/java/chess/model/position/Rank.java

```diff
@@ -1,47 +1,56 @@
 package chess.model.position;
 
-import chess.model.piece.Side;
 import java.util.Arrays;
 
 public enum Rank {
-    EIGHT(8),
-    SEVEN(7),
-    SIX(6),
-    FIVE(5),
-    FOUR(4),
-    THREE(3),
-    TWO(2),
-    ONE(1);
+    EIGHT(1, "8"),
+    SEVEN(2, "7"),
+    SIX(3, "6"),
+    FIVE(4, "5"),
+    FOUR(5, "4"),
+    THREE(6, "3"),
+    TWO(7, "2"),
+    ONE(8, "1");
 
     private final int coordinate;
+    private final String name;
 
-    Rank(final int coordinate) {
+    Rank(final int coordinate, String name) {
         this.coordinate = coordinate;
+        this.name = name;
     }
 
-    public static Rank from(final int coordinate) {
+    public static Rank from(final String name) {
         return Arrays.stream(values())
-                .filter(rank -> rank.coordinate == coordinate)
+                .filter(rank -> rank.name.equals(name))
                 .findFirst()
-                .orElseThrow(() -> new IllegalArgumentException("존재하지 않는 Rank 좌표입니다."));
+                .orElseThrow(() -> new IllegalArgumentException("존재하지 않는 Rank 이름입니다."));
     }
 
-    public int minus(final Rank other) {
-        return this.coordinate - other.coordinate;
-    }
-
-    public boolean isPawnInitialRank(final Side side) {
-        if (side.isWhite()) {
-            return TWO.equals(this);
-        }
-        return SEVEN.equals(this);
+    public boolean canMove(final int step) {
+        return canMoveUp(step) && canMoveDown(step);
     }
 
     public Rank findNextRank(final int offset) {
-        final int nextCoordinate = offset + coordinate;
         return Arrays.stream(values())
-                .filter(file -> file.coordinate == nextCoordinate)
+                .filter(rank -> hasSameCoordinate(offset, rank))
                 .findFirst()
-                .orElseThrow(() -> new IllegalStateException("존재하지 않는 Rank 좌표입니다."));
+                .orElseThrow(() -> new IllegalStateException("Rank 범위를 벗어났습니다."));
+    }
+
+    private boolean hasSameCoordinate(int offset, Rank rank) {
+        return rank.coordinate == offset + coordinate;
+    }
```

Comment:
```suggestion
    private boolean hasSameCoordinate(final int offset, final Rank rank) {
        return rank.coordinate == offset + coordinate;
    }
```
**final 적용**
final 을 붙이면 재할당을 방지한다는 점이 좋긴해요.
하지만 프로젝트가 커지고 참여인원이 많아지면 유지가 잘 되지 않는 컨벤션이기도 해요.
저만 해도 그렇게 하려다가도 빼먹는 경우가 종종 발생하거든요.
그래서 어떤 곳은 final 이 있어야하는데 없어서 '이 변수는 재할당되나?' 라는 잘못된 메시지를 전달하게 되는 것 같아요 🥲

Reply:
> 저만 해도 그렇게 하려다가도 빼먹는 경우가 종종 발생하거든요.

저도 이 말에 공감되요! 당장 지금도 3번은 전부 훑었다 생각했지만 이렇게 안 붙힌 곳이 있을 줄 몰랐거든요..
적용하기로 마음 먹었다면 좀 더 철저하게 해봐야겠단 생각이 드네요